### PR TITLE
Set primary wallet & use during credential issue process

### PR DESCRIPTION
**Pull request by Devorein**
### WHY

[Card](https://app.charmverse.io/charmverse/set-primary-credential-wallet-in-my-account-919083490366847)

### CODE REVIEW

File: pages/api/wallets/primary.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+const handler = nc<NextApiRequest, NextApiResponse>({ onError, onNoMatch });
+
+handler
+  .use(requireUser)
+  .get(getPrimaryWalletController)
+  .use(requireKeys(['walletId']))
+  .put(setPrimaryWalletController);
+
+async function getPrimaryWalletController(req: NextApiRequest, res: NextApiResponse<UserWallet | null>) {
```
Comment (motechFR) [inline pages/api/wallets/primary.ts:17]: Suggestion: Let's have this under /api/profile/primary-wallet

File: pages/api/wallets/primary.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  const userId = req.session.user.id;
+
+  const primaryWallet = await prisma.userWallet.findFirst({
+    where: {
+      userId,
+      primary: true
```
Comment (motechFR) [inline pages/api/wallets/primary.ts:23]: Regarding the model, we might end up with multiple primary wallets accidentally.

Instead, how about a primaryWallet on the User model which is a relation to UserWallet, and Cascade setNull ?

File: pages/api/wallets/primary.ts (hunk 1 above)
Comment (Devorein) [reply pages/api/wallets/primary.ts:17]: Good point, updated

File: pages/api/wallets/primary.ts (hunk 2 above)
Comment (Devorein) [reply pages/api/wallets/primary.ts:23]: Very good idea. Updated app code and core schema

**Pull request merged**